features/support/terraform_helper.rb

```diff
@@ -0,0 +1,65 @@
+require 'winrm'
+require 'fileutils'
+require 'ostruct'
+
+module TerraformHelper
+  def terraform_aws_environment(agents)
+    return if $terraform_env_created
+
+    $agents = agents.hashes.map { |hash| OpenStruct.new(hash) }
+
+    generate_terraform_project($agents)
+    run_terraform
+
+    at_exit do
+      destroy_terraform
+    end
+
+    $terraform_env_created = true
+  end
+
+  def windows_agents
+    $agents.select { |a| a.platform.downcase == 'windows' }
+  end
+
+  def redhat_agents
+    $agents.select { |a| a.platform.downcase == 'redhat' }
+  end
+
+  private
+
+  def generate_terraform_project(agents, destination=Pathname.new('tmp/terraform'))
+    # Delete existing project directory
+    FileUtils.rm_rf(destination)
+
+    # Create project directory
+    FileUtils.mkdir_p(destination)
+
+    Dir.glob(File.join('ci/template/terraform', '*')).each do |path|
+      pathname = Pathname.new(path)
+      if pathname.extname == '.erb'
+        File.open(destination.join(pathname.basename.sub(/\.erb$/, '')), 'w') do |file|
+          file.puts(ERB.new(File.read(path)).result(binding))
+        end
+      else
+        FileUtils.cp(pathname, File.join(destination, pathname.basename))
+      end
+    end
+  end
+
+  def run_terraform
+    Dir.chdir('tmp/terraform') do
+      pid = spawn('terraform init && terraform apply --auto-approve -input=false')
+      Process.wait(pid)
```

Comment:
This was to allow the std output to more naturally print to the console as cucumber is running, rather than having the cuke process capture the output into a variable. This is what I found to accomplish this, but I'm open to other methods.